File: packages/model-gateway/src/main/java/com/v3agent/gateway/perm/PermissionEntity.java
package com.v3agent.gateway.perm;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

/** 权限点字典(只读,对齐 NestJS permissions 表) */
@Data
@Entity
@Table(name = "permissions")
public class PermissionEntity {

    @Id
    private Integer id;

    /** 权限码,如 dept:member / log:view / policy:manage */
    @Column(unique = true, nullable = false, length = 64)
    private String code;

    @Column(length = 32)
    private String module;

    @Column(length = 16)
    private String action;

    @Column(length = 64)
    private String name;
}

File: packages/NestJS/src/users/permission.entity.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
} from 'typeorm';

/** 按钮级权限(查看/新增/编辑/删除/导出) */
@Entity('permissions')
export class Permission {
  @PrimaryGeneratedColumn()
  id: number;

  /** 权限码,如 user:view / user:create / user:export */
  @Column({ unique: true, length: 64 })
  code: string;

  /** 所属模块,如 user / joblevel / dept / audit / asset */
  @Column({ length: 32 })
  module: string;

  /** 动作: 查看/新增/编辑/删除/导出/管理 */
  @Column({ length: 16 })
  action: string;

  /** 中文按钮名,如 查看用户 */
  @Column({ length: 64 })
  name: string;

  @CreateDateColumn()
  createdAt: Date;
}
